chains/evm/contracts/ccvs/CCTPVerifier.sol

```diff
@@ -0,0 +1,537 @@
+  error UnknownDomain(uint64 chainSelector);
+  error UnsupportedFinality(uint32 finality);
+  error ZeroAddressNotAllowed();
+
+  event DomainsSet(DomainUpdate[] domains);
+  event DynamicConfigSet(DynamicConfig dynamicConfig);
+  event StaticConfigSet(
+    address tokenMessenger, address messageTransmitterProxy, address usdcToken, uint32 localDomainIdentifier
+  );
+  event FinalityConfigSet(FinalityConfig finalityConfig);
+
+  /// @notice The static configuration.
+  // solhint-disable-next-line gas-struct-packing
+  struct StaticConfig {
+    address tokenMessenger; // The address of the token messenger.
+    address messageTransmitterProxy; // The address of the message transmitter proxy.
+    address usdcToken; // The address of the USDC token.
+    uint32 localDomainIdentifier; // The local domain identifier.
+  }
+
+  /// @notice The arguments required to update a remote domain.
+  // solhint-disable-next-line gas-struct-packing
+  struct DomainUpdate {
+    bytes32 allowedCallerOnDest; // Address allowed to call receiveMessage on the domain (i.e. the MessageTransmitterProxy).
+    bytes32 allowedCallerOnSource; // Address allowed to call depositForBurn on the domain (i.e. the TokenMessengerProxy).
+    bytes32 mintRecipientOnDest; // Address to mint USDC to on the destination chain.
+    uint32 domainIdentifier; // Unique domain ID used across CCTP.
+    uint64 chainSelector; // The corresponding CCIP destination chain selector for the domain.
+    bool enabled; // Whether or not the domain is enabled.
+  }
+
+  /// @notice Parameters for _depositForBurn (stack too deep measure).
+  // solhint-disable-next-line gas-struct-packing
+  struct DepositForBurnParams {
+    uint256 amount; // The amount of USDC to deposit for burn.
+    bytes32 receiver; // The receiver of the minted USDC on the destination chain.
+    bytes32 messageId; // The message ID of the CCIP message.
+    bytes32 allowedCaller; // The allowed caller of the message transmitter on the destination chain.
+    uint32 finality; // The finality of the CCIP message.
+    uint32 domainIdentifier; // The domain identifier of the destination chain.
+  }
+
+  /// @notice A domain is a CCTP-specific representation of a destination chain.
+  /// @dev Zero is a valid domain identifier.
+  struct Domain {
+    bytes32 allowedCallerOnDest; // Address allowed to call receiveMessage on the domain (i.e. the MessageTransmitterProxy).
+    bytes32 allowedCallerOnSource; // Address allowed to call depositForBurn on the domain (i.e. the TokenMessengerProxy).
+    bytes32 mintRecipientOnDest; // Address to mint USDC to on the destination chain.
+    uint32 domainIdentifier; // ─╮ Unique domain ID used across CCTP.
+    bool enabled; // ────────────╯ Whether or not the domain is enabled.
+  }
+
+  /// @notice Configures finality handling for this chain.
+  struct FinalityConfig {
+    uint32 defaultCCTPFinalityThreshold; // ──╮ CCTP finality threshold applied when CCIP finality is 0.
```

Comment:
Should this not be a constant?

Reply:
Responded to most finality threshold comments elsewhere. If referring to the default (2000), then yea we could make it a constant because I doubt CCTP would ever change that (though this is never stated explicitly).